enumerable.rb

```diff
@@ -0,0 +1,119 @@
+# Enumerable method
+module Enumerable
+  def my_each
+    arr = self
+    if arr.is_a?(Array)
+      (0..(arr.length - 1)).each do |i|
+        yield(arr[i])
+      end
+    end
+    arr
+  end
+
+  def my_each_with_index
+    arr = self
+    (0..(arr.length - 1)).each do |i|
+      yield(arr[i], i)
+    end
+    arr
+  end
+
+  def my_select
+    result = []
+    my_each do |item|
+      result.push(item) if yield(item)
+    end
+    puts result.to_s
+    result
+  end
+
+  def my_all?
+    my_each do |num|
+      if yield(num) == false
+        puts false
+        return false
+      end
+    end
+    puts true
+    true
+  end
```

Comment:
- #my_all? returns false if the block returns false or nil
- #my_all? when no block or argument is given returns true when none of the collection members is false or nil
- Should work as #all? when a Regex, a Class or a pattern is given as an argument